### PR TITLE
adding a macrotask quiz question in javascript

### PR DESCRIPTION
**Pull request by ht-l1**
## Summary of changes

adding a macrotask quiz question in javascript

## Checklist

- [x] I have read and followed the [contribution guidelines](https://github.com/freeCodeCamp/Developer_Quiz_Site/blob/main/CONTRIBUTING.md).
- [x] I have read through the [Code of Conduct](https://github.com/freeCodeCamp/Developer_Quiz_Site/blob/main/CODE_OF_CONDUCT.md) and agree to abide by the rules.
- [x] This PR is for one of the available issues and is not a PR for an issue already assigned to someone else.
- [x] My PR title has a short descriptive name so the maintainers can get an idea of what the PR is about.
- [x] I have provided a summary of my changes.

closes #XXXXX

### CODE REVIEW

File: src/data/javascript-quiz.ts [hunk 1]
```diff
@@ -1968,6 +1968,16 @@ const javascriptQuiz = [
     Explanation:
       "An IIFE or Immediately Invoked Function Expression, is used to declare a private variable and to avoid polluting the global environment. If a function is declared with an IIFE by using a parenthesis around it, the variable declared inside it becomes private and its value can be accessed using a getter function inside the original function.",
     Link: "https://developer.mozilla.org/en-US/docs/Glossary/IIFE"
+  },
+  {
+    Question: "In JavaScript, which is an example of macrotask?",
```
Comment (jdwilkin4) [inline src/data/javascript-quiz.ts:1973]:
```suggestion
    Question: "In JavaScript, which of the following is an example of a macrotask?",
```

File: src/data/javascript-quiz.ts [hunk 2]
```diff
@@ -1968,6 +1968,16 @@ const javascriptQuiz = [
...
+    Answer: "macrotask examples: setTimeout(), setInterval()",
```
Comment (jdwilkin4) [inline src/data/javascript-quiz.ts:1974]:
```suggestion
    Answer: "setTimeout()",
```

File: src/data/javascript-quiz.ts [hunk 3]
```diff
@@ -1968,6 +1968,16 @@ const javascriptQuiz = [
...
+    Distractor1: "promise",
+    Distractor2: "setTimeout()",
```
Comment (jdwilkin4) [inline src/data/javascript-quiz.ts:1976]: remember that distractors are for wrong answers only

```suggestion
    Distractor2: "split()",
```

**Review by jdwilkin4 — CHANGES_REQUESTED** [review]
Thank you for your contribution!

Please review the requested changes.
Once those changes have been implemented, then we can merge your PR into main. 👍

**Comment by ht-l1** [comment]
thank you @jdwilkin4 ! Just committed all your changes :)

**Review by jdwilkin4 — APPROVED** [review]
Congrats on your first contribution to our repository. 🎉
We look forward to future contributions. 👍

**Pull request merged**